replica/tablets.cc

```diff
@@ -629,7 +639,7 @@ tablet_sstable_set::create_single_key_sstable_reader(
         const sstables::sstable_predicate& predicate) const {
     // The singular partition_range start bound must be engaged.
     auto idx = group_of(pr.start()->value().token());
-    const auto& set = _sstable_sets[idx];
+    const auto& set = find_sstable_set(idx);
```

Comment:
if `find_sstable_set` always returns an engaged set, the condition below is moot and can be removed.